### PR TITLE
sys-devel/dwz,dev-util/debugedit: fix building on musl

### PR DESCRIPTION
**Pull request by lanodan**
Closes: https://bugs.gentoo.org/829239
Closes: https://bugs.gentoo.org/902197

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-util/debugedit/files/debugedit-5.1-musl-error.h-fix.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# musl doesn't provide error.h as a result debugedit is failing to build on
```
Comment (mgorny) [inline dev-util/debugedit/files/debugedit-5.1-musl-error.h-fix.patch:1]: Was this submitted upstream?

**Comment by lanodan** [comment]
>@mgorny commented on this pull request.
>> @@ -0,0 +1,54 @@
>+# musl doesn't provide error.h as a result debugedit is failing to build on
>
>Was this submitted upstream?

Doesn't seems to have been, at least https://inbox.sourceware.org/debugedit/?q=musl doesn't returns it and it's been 2 years since.

So I sent it: ***@***.***/T/#u

**Review by mgorny — CHANGES_REQUESTED** [review]
I see upstream raises concerns regarding the patch today, so let's wait till it's all settled.

File: sys-libs/error-standalone/error-standalone-1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (thesamesam) [inline sys-libs/error-standalone/error-standalone-1.0.ebuild:4]: 8 for new ebuilds, please.

File: sys-libs/error-standalone/error-standalone-1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+DESCRIPTION="standalone <error.h> implementation intended for musl"
+HOMEPAGE="https://hacktivis.me/git/error-standalone/"
+SRC_URI="https://hacktivis.me/releases/error-standalone/${P}.tar.gz"
```
Comment (thesamesam) [inline sys-libs/error-standalone/error-standalone-1.0.ebuild:8]: With regard to error-standalone itself: AFAIK, there's implementations which require a library. You've thankfully killed that requirement. But your version doesn't install a library at all which breaks compatibility unnecessarily. Add a .pc file and install an empty library?

Comment (thesamesam) [reply sys-libs/error-standalone/error-standalone-1.0.ebuild:8]: Also, the `error-standalone` commit (adding it) should have some detail on the background in the commit message.

**Comment by lanodan** [comment]
> @thesamesam commented on this pull request.
>With regard to error-standalone itself: AFAIK, there's implementations which require a library. You've thankfully killed that requirement. But your version doesn't install a library at all which breaks compatibility unnecessarily. Add a .pc file and install an empty library?

Do you have an example of such library (which I guess is what the .pc is for)? At least I couldn't find one in Gentoo, while Alpine and Void Linux takes a similar (but somewhat incomplete) header-only approach.

**Comment by thesamesam** [comment]
I was just double-checking and you're right. I'm sorry for the confusion! I can't find any such examples.

**Review by mgorny — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**